File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h> //Rounding
#include <ctype.h> //checking for character or white space

int main(void)
{
    string s = get_string("Text: ");
    int ct_let = 0, ct_words = 1, ct_sent = 0;

    for(int i = 0, n = strlen(s); i<n ; i++)
    {
        if(isalpha(s[i]) != 0)                      //isalpha returns non 2zero if alphabetical (Counting Letters)
        {
            ct_let++;
        }
        if(isspace(s[i]) !=0)                       //isspace returns non zero is white space (Counting Words)
        {
            ct_words++;
        }
        if( s[i]==  '.' || s[i]== '!' || s[i]== '?') // Checking if char is '.' or '!' or '?' (Counting Sentences)
        {
            ct_sent++;
        }
    }
    //Coleman-Liau Formula Application
    float L,S;
    L = ((float)ct_let/ (float)ct_words)*100;          //avg. no. of letters per 100 words
    S = ((float)ct_sent/ (float)ct_words)*100;         //avg. no of sentences per 100 words
    int index = round( (0.0588*L) - (0.296*S) - 15.8); //Coleman-Liau index
    if (index<1)
    {
        printf("Before Grade 1\n");
    }
    else if(index<16)
    {
        printf("Grade %i\n",index);
    }
    else
    {
        printf("Grade 16+\n");
    }



}

File: readability.py
import cs50
from cs50 import get_string

text = get_string("Text: ")
ct_let = 0  # counter of letters
ct_words = 1  # counter of words
ct_sent = 0  # counter of sentences

# iterating through each character in text
for char in text:
    # counting number of letters
    if char.isalpha() == True:
        ct_let += 1
    # counting number of white spaces to find no. of words
    if char.isspace() == True:
        ct_words += 1
    # counting number of sentences
    if char == '.' or char =='!' or char == '?':
        ct_sent += 1

# Coleman-Liau Formula Application
L = float((float(ct_let) / float(ct_words)) * 100)
S = float((float(ct_sent) / float(ct_words)) * 100)
index = round((0.0588 * L) - (0.296 * S) - 15.8)  # Coleman-Liau index
# Checking index and printing appropriate grade
if index < 1:
    print("Before Grade 1")
elif index < 16:
    print(f"Grade {index}")
else:
    print("Grade 16+")